board.py

```diff
@@ -2,79 +2,170 @@
-        """doc"""
-        other_snake_ids = [snake_id
-                           for snake_id in data["snakes"]
-                           if snake_id != data["you"]]
-        return {snake["id"]: Snake(snake)
-                for snake in data["snakes"]
-                if snake["id"] in other_snake_ids}
-
-    def build_board(self):
-        """doc"""
-        board = [[Entity.EMPTY for _ in range(self.width + 1)] for _ in range(self.height + 1)]
-        for x, y in self.food:
-            board[y][x] = Entity.FOOD
-
-        for x, y in self.my_snake.body:
-            board[y][x] = Entity.MY_BODY
-
-        head_x, head_y = self.my_snake.head
-        board[head_y][head_x] = Entity.MY_HEAD
-
-        for other_snake in self.other_snakes:
-            for x, y in other_snake.body:
-                board[y][x] = Entity.OTHER_BODY
-
-            head_x, head_y = other_snake.head
-            board[head_y][head_x] = Entity.OTHER_HEAD
-        return board
-
-    def within_bounds(self, cell):
-        """doc"""
-        return 0 <= cell[0] <= self.width and 0 <= cell[1] <= self.height
-
-    def move_snake(self, snake_id, move):
+        self.food = self.parse_food(data["food"])
+        self.snakes = self.parse_snakes(data["snakes"])
+        self.dead_snakes = self.parse_dead_snakes(data["dead_snakes"])
+        self.build_grid()
+
+    def parse_food(self, food_data):
+        """Parse food from food_data."""
+        return set([Cell(*food) for food in food_data])
+
+    def parse_snakes(self, snakes_data):
+        """Parse snakes from snakes_data."""
+        return {snake_data["id"]: Snake(snake_data)
+                for snake_data in snakes_data}
+
+    def parse_dead_snakes(self, dead_snakes_data):
+        """Parse dead snakes from dead_snakes_data."""
+        return {dead_snake_data["id"]: Snake(dead_snake_data)
+                for dead_snake_data in dead_snakes_data}
+
+    def build_grid(self):
+        """Build a grid of EntityId that represents the current board state."""
+        self.board = [[EntityId.EMPTY for _ in range(self.width)] for _ in range(self.height)]
+
+        # Mark food
+        for cell in self.food:
+            self._set_entity_at_cell(cell, EntityId.FOOD)
+
+        # Mark live snakes
+        for cell in self.snakes.body:
+            self._set_entity_at_cell(cell, EntityId.BODY)
+        self._set_entity_at_cell(self.snakes.head, EntityId.HEAD)
+
+    def is_snake_id_at_cell(self, snake_id, cell):
+        """Return True if Entity at Cell is a Snake and has snake_id."""
+        for snake in self.snakes:
+            if snake.id is snake_id:
+                return cell in snake.head + snake.body
+        return False
+
+    def get_entity_at_cell(self, cell):
+        """Get EntityId at Cell."""
+        return self.get_entity_at_pos(cell.x_pos, cell.y_pos)
+
+    def get_entity_at_pos(self, x_pos, y_pos):
+        """Get EntityId at position."""
+        return self.board[y_pos - 1][x_pos]
```

Comment:
Can there be multiple entities at a single position?

Reply:
Or will this be resolved when fns like this are called?

Reply:
They will be resolved.